app/controllers/posts_controller.rb

```diff
@@ -9,7 +10,8 @@ def new
   end
 
   def create
-    @post = Post.new(post_params)
+    current_user
```

Comment:
ah, are you calling this so that `@current_user` is made available for use in the next line? if that's the case, then you should just change the line below to
```rb
@post = current_user.posts.build(post_params)
```